src/webportal/config/webpack.common.js

```diff
@@ -17,67 +17,68 @@
 
 
 // module dependencies
-const fs = require('fs');
 const path = require('path');
 const webpack = require('webpack');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 const CopyWebpackPlugin = require('copy-webpack-plugin');
-const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const MonacoWebpackPlugin = require('monaco-editor-webpack-plugin');
-const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const TerserPlugin = require('terser-webpack-plugin');
 const markedConfig = require('./marked.config');
 const helpers = require('./helpers');
 
 const title = 'Platform for AI';
 const version = require('../package.json').version;
 const FABRIC_DIR = path.resolve(__dirname, '../src/app/job/job-view/fabric');
 
+const devMode = process.env.NODE_ENV !== 'production';
 
-const htmlMinifierOptions = {
-  collapseWhitespace: true,
-  html5: true,
-  minifyCSS: true,
-  minifyJS: true,
-  removeComments: true,
-  removeEmptyAttributes: true,
-  removeTagWhitespace: true
-};
+function generateHtml(opt) {
+  return new HtmlWebpackPlugin(Object.assign({
+    title: title,
+    version: version,
+    template: './src/app/layout/layout.component.ejs',
+    minify: {
+      collapseWhitespace: true,
+      html5: true,
+      removeComments: true,
+      removeEmptyAttributes: true,
+      removeTagWhitespace: true,
+    },
+  }, opt));
+}
 
 const config = (env, argv) => ({
   entry: {
-    index: './src/app/index.js',
-    layout: './src/app/layout/layout.component.js',
-    register: './src/app/user/user-register/user-register.component.js',
-    userView: './src/app/user/user-view/user-view.component.js',
-    login: './src/app/user/user-login/user-login.component.js',
-    changePassword: './src/app/user/change-password/change-password.component.js',
-    dashboard: './src/app/dashboard/dashboard.component.js',
-    submit: './src/app/job/job-submit/job-submit.component.js',
-    jobList: ['babel-polyfill', './src/app/job/job-view/fabric/job-list.jsx'],
-    jobDetail: ['babel-polyfill', './src/app/job/job-view/fabric/job-detail.jsx'],
-    virtualClusters: './src/app/vc/vc.component.js',
-    services: './src/app/cluster-view/services/services.component.js',
-    hardware: './src/app/cluster-view/hardware/hardware.component.js',
-    hardwareDetail: './src/app/cluster-view/hardware/hardware-detail.component.js',
-    k8s: './src/app/cluster-view/k8s/k8s.component.js',
-    docs: './src/app/job/job-docs/job-docs.component.js',
-    jobSubmit: './src/app/marketplace/job-submit/job-submit.component.js',
-    changeGitHubPAT: './src/app/user/change-github-pat/change-github-pat.component.js',
-    howToConfigGitHubPAT: './src/app/user/how-to-config-github-pat/how-to-config-github-pat.component.js',
-    plugin: './src/app/plugin/plugin.component.js',
-
+    'index': './src/app/index.js',
```

Comment:
why use single queto in ```index```? I saw the [official doc](https://webpack.js.org/concepts/entry-points) and did not see this convention.

Reply:
https://eslint.org/docs/rules/quote-props
This rule is enabled in our eslint config file.

Reply:
got it